### PR TITLE
Make knife verbose if logging level is DEBUG or INFO

### PR DESCRIPTION
**Pull request by ifel**
Fix for https://github.com/facebook/taste-tester/issues/122

taste-tester uses between-meals to run knife commands. If it's verbose (INFO or DEBUG), make knife verbose too to simplify debug.

### CODE REVIEW

File: lib/between_meals/knife.rb [hunk 1]
```diff
@@ -39,6 +39,14 @@ def initialize(opts = {})
       @config = opts[:config] ||
                 "#{@home}/.chef/knife-#{@user}-taste-tester.rb"
       @knife = opts[:bin] || 'knife'
+      @knife_verb_option = ''
+      unless @logger.nil?
+        if @logger.level == Logger::DEBUG
+          @knife_verb_option = "-VV"
+        elsif @logger.level == Logger::INFO
```
Comment (jaymzh) [inline lib/between_meals/knife.rb]: isn't it `INFO` by default? I know in grocery-delivery its `WARNING` by default, but i think in tt it's `INFO`

File: lib/between_meals/knife.rb [hunk 2]
```diff
@@ -161,7 +169,7 @@ def databag_delete(databags)
         databags.group_by(&:name).each do |dbname, dbs|
           dbs.each do |db|
             exec!("#{@knife} data bag delete #{dbname} #{db.item}" +
-                    " --yes -c #{@config}", @logger)
+                    " --yes #{@knife_verb_option} -c #{@config}", @logger)
```
Comment (jaymzh) [inline lib/between_meals/knife.rb]: notetoself: abstract out the common parts of calling knife....

File: lib/between_meals/knife.rb (hunk 1 above)
Comment (ifel) [reply lib/between_meals/knife.rb]: In TT it's WARNING as well

Comment (ifel) [reply lib/between_meals/knife.rb]: If it's fine, LMK, I will drop this pull request and publish another one with fixed linters.

Comment (jaymzh) [reply lib/between_meals/knife.rb]: Good call. Then yes, this is perfect.

Don't drop the PR, just update it :)

Comment (ifel) [reply lib/between_meals/knife.rb]: Updated it, but circle CI does not allow me to restart the tests, as I have no rights.

Comment (jaymzh) [reply lib/between_meals/knife.rb]: it re-ran them. your lint is still failing. :)

Comment (jaymzh) [reply lib/between_meals/knife.rb]: it'll re-run them on every push (including force-pushes)

Comment (ifel) [reply lib/between_meals/knife.rb]: Fixed now. I did not know it supports force pushes.

File: lib/between_meals/knife.rb [hunk 3]
```diff
@@ -55,7 +63,8 @@ def initialize(opts = {})
     def role_upload_all
       if File.exists?(@role_dir)
         roles = File.join(@role_dir, "*.#{@role_type}")
-        exec!("#{@knife} role from file #{roles} -c #{@config}", @logger)
+        exec!("#{@knife} role from file #{roles} #{@knife_verb_option} " +
```
Comment (jaymzh) [inline lib/between_meals/knife.rb]: I'm surprised this passes lint, usually it wants these aligned, ala:
```
exec!("stuff .... " +
      "otherstuff")
```

**Comment by ifel** [comment]
Thank you for reviewing and hints!

**Comment by jaymzh** [comment]
We may want to do this differently. I just tested this, and to get the ACTUAL file with an issue you need *three* `V`s. So maybe info is VV and debug is VVV?

**Pull request merged**